src/include/mpir_pmi.h

```diff
@@ -45,12 +45,14 @@ const char *MPIR_pmi_job_id(void);
 
 /* PMI wrapper utilities */
 
-/* * barrier or kvs fence. "domain" is a hint for efficiency (eg PMIx) */
-int MPIR_pmi_barrier(MPIR_PMI_DOMAIN domain);
+/* * barrier or kvs fence. */
+int MPIR_pmi_barrier(void);
+/* * barrier over local set. More efficient for PMIx. Same as MPIR_pmi_barrier for PMI1/2. */
+int MPIR_pmi_barrier_local(void);
 /* * put, to global domain */
-int MPIR_pmi_kvs_put(char *key, char *val);
+int MPIR_pmi_kvs_put(const char *key, const char *val);
```

Comment:
To me this commit seems to mix two things in one (barrier change and const). I would split it into two patches.

Reply:
Thanks for the review. I'll make the change.